src/test/java/automationpractice/com/pages/ProductDetailsPage.java

```diff
@@ -0,0 +1,251 @@
+    private WebElementFacade clickOnFacebookButton;
+
+    @FindBy(className = "btn-google-plus")
+    private WebElementFacade clickOnGooglePlusButton;
+
+    @FindBy(className = "btn-pinterest")
+    private WebElementFacade clickOnPinteresButton;
+
+    private int currentQuantityInProductDetailsPage;
+
+    private String currentSizeInProductDetailsPage;
+
+    @FindBy(xpath = "//*[@id = 'color_to_pick_list']//*[contains(@id,'color')]")
+    private List<WebElementFacade> listOfColorsAndSizes;
+
+    private String currentColorInProductDetailsPage;
+
+    public ProductDetailsPage() {
+    }
+
+    public void clickOnAddToCartButton() {
+        clickOnAddToCartButton.click();
+    }
+
+    public void clickProceedToCheckoutButton() {
+        clickProceedToCheckoutButton.click();
+    }
+
+    private void clickPlusQuantityButtonOnProductDetailsPage() {
+        clickPlusQuantityButtonOnProductDetailsPage.click();
+    }
+
+    private void clickMinusQuantityButtonOnProductDetailsPage() {
+        clickMinusQuantityButtonOnProductDetailsPage.click();
+    }
+
+    public int clickPlusAndMinusButtonGivenNumberOfTimesOnProductDetailsPage(int plus, int minus) {
+        currentQuantityInProductDetailsPage = 1;
+        for (int i = 0; i < plus; i++) {
+            waitABit(1000);
+            clickPlusQuantityButtonOnProductDetailsPage();
+            currentQuantityInProductDetailsPage++;
+        }
+        for (int i = 0; i < minus; i++) {
+            waitABit(1000);
+            clickMinusQuantityButtonOnProductDetailsPage();
+            currentQuantityInProductDetailsPage--;
+        }
+        return currentQuantityInProductDetailsPage;
+    }
+
+    public int getCurrentQuantitySetUpInProductDetailsPage() {
+        return currentQuantityInProductDetailsPage;
+    }
+
+    public String changeSizeToGivenSizeInProductDetailsPage(String sizeWrittenInLargeLetters) {
+        currentSizeInProductDetailsPage = sizeWrittenInLargeLetters;
+        switch (sizeWrittenInLargeLetters) {
+            case "S":
+                selectSSize.click();
+                break;
+            case "M":
+                selectMSize.click();
+                break;
+            case "L":
+                selectLSize.click();
+                break;
+            default:
+                selectSSize.click();
+                break;
+        }
+        return sizeWrittenInLargeLetters;
+    }
+
+    public String getCurrentSizeSetUpInProductDetailsPage() {
+        return currentSizeInProductDetailsPage;
+    }
+
+    private ArrayList getAvailableColorsInProductDetailsPage() {
+        ArrayList<String> listOfColors = new ArrayList<>();
+
+        for (WebElementFacade row : listOfColorsAndSizes) {
+            String[] split = row.toString().split("<a id='color_(.*)' name='");
+            String[] color = split[1].split("'>(.*)");
+            listOfColors.add(color[0]);
+        }
+        return listOfColors;
+    }
```

Comment:
Tutaj lepiej na każdym elemencie list listOfColorsAndSizes wywołać metodę getAttribute("name");
Czyli zamiast całego tego for można napisać
return listOfColorsAndSizes.stream().map(row -> row.getAttribute("name")).collect(Collectors.toList());

Reply:
Nie rozumiem po co switch. Nie możesz po prostu
currentSize = size.toUpperCase();
selectSize(currentSize);
plus ta metoda nie powinna zwracać Stringa